src/Elliptic/FirstOrderOperator.hpp

```diff
@@ -60,27 +60,29 @@ struct FirstOrderSourcesImpl<Dim, tmpl::list<PrimalFields...>,
       const Variables<db::wrap_tags_in<
           ::Tags::Flux, VarsTags, tmpl::size_t<Dim>, Frame::Inertial>>& fluxes,
       const SourcesArgs&... sources_args) noexcept {
+    // Compute primal sources. They may depend on the auxiliary variables in
+    // addition to the primal variables. However, we pass the fluxes instead of
+    // the auxiliary variables to the source computer as an optimization so they
+    // don't have to be re-computed.
+    sources->initialize(vars.number_of_grid_points(), 0.);
+    SourcesComputer::apply(
+        make_not_null(&get<::Tags::Source<PrimalFields>>(*sources))...,
+        sources_args..., get<PrimalFields>(vars)...,
+        get<::Tags::Flux<PrimalFields, tmpl::size_t<Dim>, Frame::Inertial>>(
+            fluxes)...);
     // Set auxiliary field sources to the auxiliary field values to begin with.
     // This is the standard choice, since the auxiliary equations define the
-    // auxiliary variables. Source computers can adjust or add to these sources.
+    // auxiliary variables. Source computers can add to these sources.
```

Comment:
SourceComputer::apply is _before_ setting auxiliary field sources.  So how can the source computer add to the sources?  Please clarify comment.

Reply:
There's another `SourceComputer::apply` below, which adds the auxiliary sources. The one above adds the primal sources. Do you think that's unclear?

Reply:
[optional] maybe instead of `apply` the functions could be called `add_primal_sources` and `add_auxiliary_sources`? I would find that easier to follow, and would also make the concrete implementations easier to read, I suspect. I generally think we should avoid function overloading like this if it's possible to avoid use confusion.

Reply:
Agreed, though here I originally named the functions `apply` so they work with `db::apply` and friends. That isn't very important anymore, so I'll consider changing the names (not in this PR, because it involves more code and the flux computer functions should be renamed too)